### ISSUE TITLE
Accessing class probabilities in c++

### ISSUE THREAD

**Issue by hyungwonchoi** [BVLC/caffe#753]
I am currently working on webcam demo of ImageNet pretrained network using c++. 
I want to access each class probability, which is the output of the softmax layer.

The definition of the softmax layer in the test prototxt is as follows.

```
layers {
  name: "prob"
  type: SOFTMAX
  bottom: "fc8"
  top: "prob"
}
```

I am trying to access the output of the softmax like

``` c++
const shared_ptr<Blob<float> >& probs = caffe_test_net.blob_by_name("prob");
const float* probs_out = probs->cpu_data();
```

However, when I print out probs_out[0] to probs_out[999], they all contain 0. 
How can I access the class probability ? 

Any helps would be appreciated.

Hyungwon

**Comment by deepaksharma36** [comment]
If I used batch size >1 then how I can access probabilities distribution for each sample of the batch?
